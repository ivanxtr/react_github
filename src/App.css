body, html {
  width: 100%;
  height: 100%;
}
#root {
  width: 100%;
  height: 100%;
}
.App {
  display: flex;
  justify-content: flex-start;
  align-items: center;
  height: 100%;
  flex-direction: column;
}
.bugs {
  width: 40%;
  display: flex;
  justify-content: center;
  align-items: flex-start;
  flex-wrap: wrap;
  flex-direction: column;
}
.search {
  margin: 80px 0 0 0;
  width: 40%;
  padding: 10px;
  margin-right: 10px;
  background-color: #eeeeee;
  border: solid 1px #bbbbbb;
  padding: 15px;
}
.bugs > .item {
  padding: 10px;
  margin: 10px 0;
  width: 96%;
  background-color: #f1fffd;
  border: solid 2px #b7deff;
}
