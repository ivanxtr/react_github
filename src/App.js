import React, { useState, useEffect } from 'react'
import './App.css';
import Search from './components/searchComponent'
import { Promises, Filter } from './factory/index'
import Content from './components/content'

function App() {

  const [data, setData] = useState([])
  const [cacheData, setCacheData] = useState([])

  // calling the API
  const apicall = async () => {
    const call = await new Promises('https://api.github.com/repos/facebook/react/issues')
    const result = await call.getReactBugs()
    setCacheData(result)
    return setData(result)
  }

  const filterData = async (query) => {
    const cloneData = JSON.parse(JSON.stringify(cacheData))
    const filter = new Filter(cloneData, query)
    const response = await filter.filterByTitle()
    return setData(response)
  }

  // rendering thte initial data
  useEffect(() => {
    apicall()
  },[])

  return (
    <div className="App">
      <Search filterData={filterData} />
      <Content data={data} />
    </div>
  );
}

export default App;
