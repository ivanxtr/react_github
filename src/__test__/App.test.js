import React from 'react';
import ReactDOM from 'react-dom';
import { create } from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import App from '../App';

Enzyme.configure({ adapter: new Adapter() });

describe('App Component', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('snapshot render', () => {
    const component = create(<App />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('should render the proper structure', () => {
    const wrapper = mount(<App />)
    expect(wrapper.find('.App')).toHaveLength(1);
    expect(wrapper.find('input')).toHaveLength(1);
    expect(wrapper.find('.bugs')).toHaveLength(1);
  })
})
