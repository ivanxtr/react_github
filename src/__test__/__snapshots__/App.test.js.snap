// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component snapshot render 1`] = `
<div
  className="App"
>
  <input
    className="search"
    onChange={[Function]}
    placeholder="Search"
    type="search"
  />
  <div
    className="bugs"
  />
</div>
`;
